### PR TITLE
Attach type for record arguments.

### PR DESCRIPTION
**Pull request by b4h0-c4t**
## Why
`record` arguments need `Record` type.

## What
Can use `Record` type in calling `RecordClient`.

## How to test
Modify existing `RecordClient` type test to `Record`.

## Checklist

- [x] Read [CONTRIBUTING.md](https://github.com/kintone/js-sdk/blob/master/CONTRIBUTING.md)
- [x] Updated documentation if it is required.
- [x] Added tests if it is required.
- [x] Passed `yarn lint` and `yarn test` on the root directory.

### CODE REVIEW

File: packages/rest-api-client/src/client/RecordClient.ts [hunk 1]
```diff
@@ -17,6 +17,32 @@ const ADD_RECORDS_LIMIT = 100;
 const UPDATE_RECORDS_LIMIT = 100;
 const DELETE_RECORDS_LIMIT = 100;
 
+type Entity = { code: string; name: string };
```
Comment (shintaroNagata) [inline packages/rest-api-client/src/client/RecordClient.ts:20]: In parameter, `name` property is cannot specified.

File: packages/rest-api-client/src/client/RecordClient.ts [hunk 2]
```diff
@@ -17,6 +17,32 @@ const ADD_RECORDS_LIMIT = 100;
...
+      | {
+          [fieldCode: string]: {
+            value:
+              | number
+              | string
+              | string[]
+              | null
+              | FileInformation[]
+              | Entity[];
+          };
```
Comment (shintaroNagata) [inline packages/rest-api-client/src/client/RecordClient.ts:41]: The `value` parameter of a Subtable field has Array type value, and their elements can have `id` property.
Also, `value` property can be optional when `id` property is specified.

File: packages/rest-api-client/src/client/RecordClient.ts [hunk 3]
```diff
@@ -93,7 +130,9 @@ export class RecordClient {
     }
     return this.addRecord({
       app,
-      record: { ...record, [updateKey.field]: { value: updateKey.value } },
+      record: Object.assign(record, {
```
Comment (shisama) [inline packages/rest-api-client/src/client/RecordClient.ts:133]:
```suggestion
      record: Object.assign({}, record, {
```

File: packages/rest-api-client/src/client/RecordClient.ts [hunk 4]
```diff
@@ -17,35 +17,37 @@ const ADD_RECORDS_LIMIT = 100;
...
-              }
-            | {
-                id: number | string;
-                value?: SubtableFieldValue;
-              };
-        };
+      | FileInformationForParametor[]
+      | EntityForParametor
+      | EntityForParametor[]
+      | SubtableForParametor;
```
Comment (shisama) [inline packages/rest-api-client/src/client/RecordClient.ts:50]: @b4h0-c4t I found some misspellings. Please look for others.

```suggestion
      | FileInformationForParameter[]
      | EntityForParameter
      | EntityForParameter[]
      | SubtableForParameter;
```

**Pull request merged**